console/console-init/ui/src/modules/address-space/dialogs/CreateAddressSpace/Configuration.tsx

```diff
@@ -139,15 +139,14 @@ export const Configuration: React.FunctionComponent<IConfiguration> = ({
     if (type) {
       planOptions =
         addressSpacePlans
+          .filter(plan => plan.spec.addressSpaceType === type)
           .map(plan => {
-            if (plan.spec.addressSpaceType === type) {
-              return {
-                value: plan.metadata.name,
-                label: plan.spec.displayName || plan.metadata.name,
-                description:
-                  plan.spec.shortDescription || plan.spec.longDescription
-              };
-            }
+            return {
+              value: plan.metadata.name,
+              label: plan.spec.displayName || plan.metadata.name,
+              description:
+                plan.spec.shortDescription || plan.spec.longDescription
+            };
           })
           .filter(plan => plan !== undefined) || [];
```

Comment:
Above we have already filter, can we try to add this filter check in above filter and remove this  filter.
addressSpacePlans.filter(plan => plan.spec.addressSpaceType === type)==> addressSpacePlans.filter(plan => plan.spec.addressSpaceType === type && plan !== undefined)

Reply:
Done

Reply:
Had to revert the changes due to test case failure.